xarray/plot/plot.py

```diff
@@ -240,6 +240,47 @@ def _infer_line_data(darray, x, y, hue):
     return xplt, yplt, hueplt, xlabel, ylabel, huelabel
 
 
+def _infer_scatter_data(ds, x, y, hue):
+    dvars = set(ds.data_vars.keys())
+    error_msg = (' must be either one of ({0:s})'
+                 .format(', '.join(dvars)))
+
+    if x not in dvars:
```

Comment:
what if `x = None` or `y=None` or both are None. If there are only 2 datavars then we can guess the other if either `x` or `y` are provided.

Reply:
I don't know about that.
Another option is to allow multiple pairs: A vs. B and C vs. D on the same axis with different hues, or A vs. {B, C, D, E} where the other DataArrays are different hues/different columns?

Reply:
> Another option is to allow multiple pairs: A vs. B and C vs. D on the same axis with different hues

This should be doable with `facetgrid.map`?

> A vs. {B, C, D, E} where the other DataArrays are different hues/different columns?

This seems like it would make the API too complicated for not much gain.

Reply:
> what if x = None or y=None or both are None. If there are only 2 datavars then we can guess the other if either x or y are provided.

At the least, we should raise an error when  `x` and `y` are `None`. My initial thought was that `DataArray.plot()` does some autoguessing but `Dataset.plot()` can only do this if there are 2 compatible `Dataset.variables`. Maybe that's a special case not worth worrying about and we should just ask users to be explicit about what they want for `x` and `y`